### PR TITLE
Update AA summary logics & Mutation Effect

### PR DESCRIPTION
**Pull request by zhx828**
- Update logics on generating alternative allele summary

- Update mutation effect description based on alteration

    If we do not have mutation effect for the queried alteration but alternative allele or other relevant alteration does, we add a leading sentence to it.

This fixes https://github.com/oncokb/oncokb-pipeline/issues/134

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/oncokb/util/SummaryUtils.java [hunk 1]
```diff
@@ -476,28 +520,141 @@ private static String enrichGeneEvidenceDescription(EvidenceType evidenceType, G
...
             String altStr = getGeneMutationNameInVariantSummary(alteration.getGene(), referenceGenome, queryHugoSymbol, alteration.getAlteration());
-            sb.append("The " + altStr + " has not been functionally or clinically validated.");
+            Oncogenicity referredOncogenicity = getRefAlleleOncogenicityBasedOnAltAlleles(allelesOncogenicities);
+            if (referredOncogenicity == null) {
+                referredOncogenicity = Oncogenicity.UNKNOWN;
+            }
+            if (isVus) {
+                sb.append(getVUSSummary(alteration, altStr, true));
+            } else {
+                sb.append("The " + altStr + " has not specifically been reviewed by the OncoKB team.");
```
Comment (calvinlu3) [inline core/src/main/java/org/mskcc/cbio/oncokb/util/SummaryUtils.java:626]: Is the word `mutation` after the `altStr` not needed anymore? I see in the doc that some places don't have it, but others do.

File: core/src/test/java/org/mskcc/cbio/oncokb/util/MutationEffectParameterizedTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    public void testSummary() throws Exception {
+        Query query = new Query();
+        query.setHugoSymbol(gene);
+        query.setAlteration(variant);
+
+        IndicatorQueryResp resp = IndicatorUtils.processQuery(query, null, false, null);
+        String _query = gene + " " + variant;
+        String _mutationEffect = resp.getMutationEffect().getKnownEffect();
+        String _description = resp.getMutationEffect().getDescription();
+//        System.out.println("New: " + gene + "&&" + variant + "&&" + _mutationEffect + "&&" + _description);
```
Comment (calvinlu3) [inline core/src/test/java/org/mskcc/cbio/oncokb/util/MutationEffectParameterizedTest.java:45]: Still needed?

File: core/src/test/resources/test_variant_summaries.tsv [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+SMO	E518V	Unknown	The SMO E518V mutation has not specifically been reviewed by the OncoKB team. While SMO E518K is likely neutral and SMO E518A has been found in the context of resistance, the oncogenic effect of SMO E518V is unknown.
+ERBB2	E719A	Unknown	The ERBB2 E719A mutation has not specifically been reviewed by the OncoKB team. While ERBB2 E719G has been found in the context of resistance, the oncogenic effect of ERBB2 E719A is unknown.
+
+## Table 9.3: Leading Sentences, mainly test unknown mutation but fall under umbrella terms(Truncating Mutations, Fusion)
+### Truncating mutation in oncogene
+BRAF	T100*	Unknown	BRAF is considered an oncogene and truncating mutations in oncogenes are typically nonfunctional.
+### Truncating mutation in TSG
+BRCA1	T100*	Likely Oncogenic	BRCA1 T100* is a truncating mutation in a tumor suppressor gene, and therefore is likely oncogenic.
+### Truncating mutations in TSG that not curated
```
Comment (calvinlu3) [inline core/src/test/resources/test_variant_summaries.tsv:30]:
```suggestion
### Truncating mutations in TSG that are not curated
```

File: core/src/test/java/org/mskcc/cbio/oncokb/util/MutationEffectParameterizedTest.java (hunk 2 above)
Comment (zhx828) [reply core/src/test/java/org/mskcc/cbio/oncokb/util/MutationEffectParameterizedTest.java:45]: Yup, this is used in all tests to populate all tests so we don't have to modify individually when there are a lot of valid tests to be modified.

File: core/src/main/java/org/mskcc/cbio/oncokb/util/SummaryUtils.java (hunk 1 above)
Comment (zhx828) [reply core/src/main/java/org/mskcc/cbio/oncokb/util/SummaryUtils.java:626]: `altStr` includes `mutation` when necessary.

**Review by calvinlu3 — APPROVED** [review]
LGTM

**Pull request merged**